shiny/_docstring.py

```diff
@@ -88,25 +116,40 @@ def _(func: F) -> F:
             ex_dir_found = find_api_examples_dir(func_dir)
 
             if ex_dir_found is None:
-                raise ValueError(
+                raise FileNotFoundError(
                     f"No example directory found for {fn_name} in {func_dir} or its parent directories."
                 )
             example_dir = os.path.join(ex_dir_found, fn_name)
         else:
-            example_dir = os.path.join(func_dir, ex_dir)
+            example_dir = os.path.abspath(os.path.join(func_dir, ex_dir))
 
-        example_file = os.path.join(example_dir, app_file)
-        if not os.path.exists(example_file):
-            raise ValueError(
-                f"No example for {fn_name} found in '{os.path.abspath(example_dir)}'."
+            if not os.path.exists(example_dir):
+                raise FileNotFoundError(
+                    f"Example directory '{example_dir}' does not exist for {fn_name}."
+                )
+
+        app_file_name = app_file or "app.py"
+        try:
+            example_file = app_choose_core_or_express(
+                os.path.join(example_dir, app_file_name)
             )
+        except ExampleNotFoundException as e:
+            func_dir = get_decorated_source_directory(func).split("py-shiny/")[1]
+            if "__code__" in dir(func):
+                print(
+                    f"::warning file={func_dir},line={func.__code__.co_firstlineno}::{e}"
+                )
+            else:
+                print(f"::warning file={func_dir}::{e}")
```

Comment:
Would be better not to assume `py-shiny` is always in the path

```suggestion
            file = "shiny/" + func_dir.split("shiny/")[1]
            if "__code__" in dir(func):
                print(
                    f"::warning file={file},line={func.__code__.co_firstlineno}::{e}"
                )
            else:
                print(f"::warning file={file}::{e}")
```

Reply:
Good point, thanks! Also good catch on not needing to find `func_dir` again (at one point that was behind one of the conditionals above and I forgot I could re-use it).